File: Practica 4/Ej5/imprime_pares.c
#include <stdio.h>

void imprime_pares_intervalo(int a, int b)
{
    for (int i = b - 1; i > a; i--)
    {
        if (i % 2 != 0)
        {
            continue;
        }

        printf("%d ", i);
    }
}

int main()
{
    int a, b;

    printf("Introduce un número: ");
    scanf("%d", &a);

    printf("Introduce otro número mayor que el anterior: ");
    scanf("%d", &b);

    while (a >= b)
    {
        printf("Error: El segundo número debe ser mayor que el primero\n");

        printf("Vuelve a introducirlo: ");
        scanf("%d", &b);
    }

    imprime_pares_intervalo(a, b);
}

File: Practica 6/Ej5/imprime_pares.py
def main():
    n1 = int(input("Introduce un número: "))
    n2 = int(input("Introduce otro número: "))

    while n1 > n2:
        print("Error: El primero debe ser menor que el segundo")
        n1 = int(input("Introduce un número: "))
        n2 = int(input("Introduce otro número: "))

    imprime_pares_intervalo(n1, n2)


def imprime_pares_intervalo(n1, n2):
    for n in range(n2 - n1 - 1):
        if (n2 - n - 1) % 2 == 0:
            print(f"{n2 - n - 1} ", end="")


main()
